lib/src/mt_instance.c

```diff
@@ -90,6 +91,17 @@ int mt_instance_init(struct mtl_main_impl* impl) {
   reg_msg->uid = htonl(getuid());
   strncpy(reg_msg->hostname, u_info->hostname, sizeof(reg_msg->hostname));
 
+  /* manager will load xdp program for afxdp interfaces */
+  uint16_t num_xdp_if = 0;
+  for (int i = 0; i < p->num_ports; i++) {
+    if (p->pmd[i] == MTL_PMD_NATIVE_AF_XDP) {
```

Comment:
I guess for MTL_PMD_DPDK_AF_XDP it should also let manager to load the XDP. You can use `mtl_pmd_is_af_xdp`

Reply:
Done. Yes, we can also load our XDP program earlier than PMD.